src/Microsoft.Extensions.Options/IOptionsValidator.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Used to validate TOptions instances.
+    /// </summary>
+    /// <typeparam name="TOptions">The type of options being requested.</typeparam>
+    public interface IOptionsValidator<TOptions> where TOptions : class, new()
```

Comment:
A validator implementation can choose to aggregate the results if it wants by wrapping each action in a try/catch, so its not an issue.